### PR TITLE
gh-122767: document "new style" formatting for complexes

### PR DESCRIPTION
**Pull request by skirpichev**
* Issue: gh-122767

----
📚 Documentation preview 📚: https://cpython-previews--122848.org.readthedocs.build/

### CODE REVIEW

**Review by ericvsmith — APPROVED** [review]
This looks good to me.

File: Doc/library/string.rst [hunk 1]
```diff
@@ -588,6 +589,14 @@ The available presentation types for :class:`float` and
    |         | as altered by the other format modifiers.                |
    +---------+----------------------------------------------------------+
 
+The available presentation types for :class:`complex` are same as for
+:class:`float`.  Both real and imaginary components of a complex number are
+formatted as floating-point numbers, accordingly specified presentaion type.
+They separated by mandatory sign of the imaginary part, terminated by ``j``
+suffix.  If presentation type is missing --- result match the output of
+:func:`str` (numbers also surrounded by round parentheses, if real part is
+non-zero), altered by other format modifiers.
```
Comment (picnixz) [inline Doc/library/string.rst]:
```suggestion
The available presentation types for :class:`complex` are the same as for
:class:`float`. The real and imaginary components of a complex number are
formatted as floating-point numbers, according to the presentation type.
They are separated by a mandatory sign of the imaginary part suffixed by ``j``.
If the presentation type is not specified, the result matches the output of
:func:`str` (numbers are surrounded by round parentheses if the real part is
non-zero), possibly altered by other format modifiers.
```

**Review by picnixz — COMMENTED** [review]
I could use some help from @python/proofreaders.

File: Doc/library/string.rst (hunk 1 above)
Comment (skirpichev) [reply Doc/library/string.rst]: Does make sense for me, but I would prefer to avoid examples, as for other format types.

Comment (picnixz) [reply Doc/library/string.rst]: I've updated the suggestion.

File: Doc/library/string.rst [hunk 2]
```diff
@@ -588,6 +589,14 @@ The available presentation types for :class:`float` and
    |         | as altered by the other format modifiers.                |
    +---------+----------------------------------------------------------+
 
+The available presentation types for :class:`complex` are same as for
+:class:`float`.  Both real and imaginary components of a complex number are
+formatted as floating-point numbers, accordingly specified presentaion type.
+They separated by mandatory sign of the imaginary part, terminated by ``j``
+suffix.  If presentation type is missing --- result match the output of
+:func:`str` (numbers also surrounded by round parentheses, if real part is
```
Comment (bskinn) [inline Doc/library/string.rst]: Assuming that by 'round parentheses' you mean the `()` characters, then 'round' is unnecessary. `()` are the only symbols named 'parentheses'.

There _is_ some ambiguity for other enclosures, though, so the reflex to include the shape adjective makes sense:

 - `{}` -- 'braces', 'curly braces', or 'curly brackets'
 - `[]` -- 'brackets', 'square brackets' or (rarely) 'square braces'
 - `<>` -- 'angle brackets'

Comment (skirpichev) [inline Doc/library/string.rst]: Ok, lets keep your suggestion:
```suggestion
:func:`str` (numbers also surrounded by parentheses, if real part is
```
This pr was approved, so I'll not commit it.

FYI, current wording was copied from the complex type docs: https://docs.python.org/3.14/library/functions.html#complex

Comment (ericvsmith) [reply Doc/library/string.rst]: I think removing "round" is a good change.

Comment (bskinn) [reply Doc/library/string.rst]: Ah, @skirpichev, indeed -- sorry for dropping that unnecessary explainer on you!

File: Doc/library/string.rst [hunk 3]
```diff
@@ -588,6 +589,14 @@ The available presentation types for :class:`float` and
    |         | as altered by the other format modifiers.                |
    +---------+----------------------------------------------------------+
 
+The available presentation types for :class:`complex` are same as for
+:class:`float`.  Both real and imaginary components of a complex number are
+formatted as floating-point numbers, accordingly specified presentaion type.
```
Comment (ericvsmith) [inline Doc/library/string.rst]: Typo in "presentation". Also, I think this should be "are formatted as floating-point numbers, according to the specified presentation type."

File: Doc/library/string.rst [hunk 4]
```diff
@@ -588,6 +589,14 @@ The available presentation types for :class:`float` and
...
+They separated by mandatory sign of the imaginary part, terminated by ``j``
```
Comment (ericvsmith) [inline Doc/library/string.rst]: Should be "They are separated by the mandatory sign of the ...".

File: Doc/library/string.rst [hunk 5]
```diff
@@ -588,6 +589,14 @@ The available presentation types for :class:`float` and
...
+suffix.  If presentation type is missing --- result match the output of
```
Comment (ericvsmith) [inline Doc/library/string.rst]: Should be "the result will match".

**Review by ericvsmith — CHANGES_REQUESTED** [review]
Sorry, I should have made my previous comments into a single review.

**Comment by skirpichev** [comment]
I have made the requested changes; please review again.

File: Doc/library/string.rst [hunk 6]
```diff
@@ -588,6 +589,15 @@ The available presentation types for :class:`float` and
    |         | as altered by the other format modifiers.                |
    +---------+----------------------------------------------------------+
 
+The available presentation types for :class:`complex` are same as for
+:class:`float` (``'%'`` is not allowed).  Both real and imaginary components of
+a complex number are formatted as floating-point numbers, according to the
+specified presentation type.  They are separated by the mandatory sign of the
+imaginary part, terminated by ``j`` suffix.  If presentation type is missing
+--- the result will match the output of :func:`str` (numbers also surrounded by
+round parentheses, if real part is non-zero), altered by other format
```
Comment (ericvsmith) [inline Doc/library/string.rst]: I don't think we typically say "round parentheses", so please just use "parentheses".

**Review by ericvsmith — CHANGES_REQUESTED** [review]
Small tweak. Otherwise looks good!

File: Doc/library/string.rst (hunk 6 above)
Comment (skirpichev) [inline Doc/library/string.rst]:
```suggestion
parentheses, if real part is non-zero), altered by other format
```

File: Doc/library/string.rst [hunk 7]
```diff
@@ -588,6 +589,15 @@ The available presentation types for :class:`float` and
...
    +---------+----------------------------------------------------------+
 
+The available presentation types for :class:`complex` are same as for
+:class:`float` (``'%'`` is not allowed).  Both real and imaginary components of
+a complex number are formatted as floating-point numbers, according to the
+specified presentation type.  They are separated by the mandatory sign of the
+imaginary part, terminated by ``j`` suffix.  If presentation type is missing
+--- the result will match the output of :func:`str` (numbers also surrounded by
+parentheses, if real part is non-zero), altered by other format
+modifiers.
```
Comment (skirpichev) [inline Doc/library/string.rst]:
```suggestion
parentheses, if real part is non-zero), altered by other format modifiers.
```

File: Doc/library/string.rst (hunk 6 above)
Comment (skirpichev) [reply Doc/library/string.rst]: FYI: This wording come from complex() constructor docs.

**Comment by skirpichev** [comment]
I have made the requested changes; please review again.

File: Doc/library/string.rst [hunk 8]
```diff
@@ -588,6 +589,14 @@ The available presentation types for :class:`float` and
    |         | as altered by the other format modifiers.                |
    +---------+----------------------------------------------------------+
 
+The available presentation types for :class:`complex` are same as for
+:class:`float` (``'%'`` is not allowed).  Both real and imaginary components of
+a complex number are formatted as floating-point numbers, according to the
+specified presentation type.  They are separated by the mandatory sign of the
+imaginary part, terminated by ``j`` suffix.  If presentation type is missing
+--- the result will match the output of :func:`str` (numbers also surrounded by
```
Comment (ericvsmith) [inline Doc/library/string.rst]: What are the "---" for?

Comment (skirpichev) [reply Doc/library/string.rst]: Hmm, probably ',' is better, indeed.

Comment (skirpichev) [inline Doc/library/string.rst]:
```suggestion
imaginary part, terminated by ``j`` suffix.  If presentation type is missing,
the result will match the output of :func:`str` (numbers also surrounded by
```

**Review by ericvsmith — APPROVED** [review]

File: Doc/library/string.rst [hunk 9]
```diff
@@ -588,6 +589,14 @@ The available presentation types for :class:`float` and
    |         | as altered by the other format modifiers.                |
    +---------+----------------------------------------------------------+
 
+The available presentation types for :class:`complex` are same as for
+:class:`float` (``'%'`` is not allowed).  Both real and imaginary components of
+a complex number are formatted as floating-point numbers, according to the
+specified presentation type.  They are separated by the mandatory sign of the
+imaginary part, terminated by ``j`` suffix.  If presentation type is missing,
+the result will match the output of :func:`str` (numbers also surrounded by
+parentheses, if real part is non-zero), altered by other format modifiers.
```
Comment (picnixz) [inline Doc/library/string.rst]:
```suggestion
The available presentation types for :class:`complex` are the same as those for
:class:`float` (``'%'`` is not allowed).  Both the real and imaginary components
of a complex number are formatted as floating-point numbers, according to the
specified presentation type.  They are separated by the mandatory sign of the
imaginary part, the latter being terminated by a ``j`` suffix.  If the presentation
type is missing, the result will match the output of :func:`str` (complex numbers with
a non-zero real part are also surrounded by parentheses), possibly altered by
other format modifiers.
```

**Review by picnixz — COMMENTED** [review]
A final nitpick on wording.

**Comment by skirpichev** [comment]
CC @ericvsmith, are you waiting for the second opinion?  (BTW, I'm ok above suggestion by @picnixz)

**Pull request merged**